### PR TITLE
mpvScripts.memo: init at 0-unstable-2023-10-31

### PR DESCRIPTION
**Pull request by purrpurrn**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by purrpurrn** [comment]
@ofborg eval

File: pkgs/applications/video/mpv/scripts/memo.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  installPhase = ''
+    mkdir -p $out/share/mpv/scripts
+    cp -r memo.lua $out/share/mpv/scripts
+  '';
+
+  meta = with lib; {
+    description = "A recent files menu for mpv.";
+    homepage = "https://github.com/po5/memo";
+    license = licenses.gpl3;
```
Comment (dotlambda) [inline pkgs/applications/video/mpv/scripts/memo.nix:29]: `gpl3Only` or `gol3Plus`?

File: pkgs/applications/video/mpv/scripts/memo.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  passthru.scriptName = "memo.lua";
+
+  installPhase = ''
+    mkdir -p $out/share/mpv/scripts
+    cp -r memo.lua $out/share/mpv/scripts
+  '';
+
+  meta = with lib; {
+    description = "A recent files menu for mpv.";
```
Comment (dotlambda) [inline pkgs/applications/video/mpv/scripts/memo.nix:27]:
```suggestion
    description = "A recent files menu for mpv";
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

File: pkgs/applications/video/mpv/scripts/memo.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{ fetchFromGitHub
+, buildLua
+, lib
+, unstableGitUpdater
+}:
+
+buildLua {
+  pname = "memo";
+  version = "unstable-2023-10-31";
```
Comment (dotlambda) [inline pkgs/applications/video/mpv/scripts/memo.nix:9]:
```suggestion
  version = "0-unstable-2023-10-31";
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

File: pkgs/applications/video/mpv/scripts/memo.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  src = fetchFromGitHub {
+    owner = "po5";
+    repo = "memo";
+    rev = "6f2b036ef860e130ea584657389e0a7ceb4d954f";
+    sha256 = "03rk1f97jblnajgwr36qz7hq1qlxrgkywc0gnq0wwg1vxiga9j4v";
```
Comment (dotlambda) [inline pkgs/applications/video/mpv/scripts/memo.nix:15]:
```suggestion
    hash = "...";
```

File: pkgs/applications/video/mpv/scripts/memo.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  };
+  passthru.updateScript = unstableGitUpdater {};
+
+  passthru.scriptName = "memo.lua";
+
+  installPhase = ''
+    mkdir -p $out/share/mpv/scripts
+    cp -r memo.lua $out/share/mpv/scripts
```
Comment (dotlambda) [inline pkgs/applications/video/mpv/scripts/memo.nix:23]:
```suggestion
    runHook preInstall

    mkdir -p $out/share/mpv/scripts
    cp -r memo.lua $out/share/mpv/scripts

    runHook postInstall
```

File: pkgs/applications/video/mpv/scripts/memo.nix (hunk 1 above)
Comment (purrpurrn) [reply pkgs/applications/video/mpv/scripts/memo.nix:33]: Sorry, I meant gpl3Only.
```suggestion
    license = licenses.gpl3Only;
```

Comment (purrpurrn) [reply pkgs/applications/video/mpv/scripts/memo.nix]: fixed in [5615f1dcc5c3a6164ef5dc5d5e017f18df632e25](https://github.com/NixOS/nixpkgs/pull/288105/commits/5615f1dcc5c3a6164ef5dc5d5e017f18df632e25)

File: pkgs/applications/video/mpv/scripts/memo.nix (hunk 4 above)
Comment (purrpurrn) [reply pkgs/applications/video/mpv/scripts/memo.nix]: fixed in [b0feaf53aefa0cfce056767e51ea658254b0fb33](https://github.com/NixOS/nixpkgs/pull/288105/commits/b0feaf53aefa0cfce056767e51ea658254b0fb33)

**Comment by purrpurrn** [comment]
@dotlambda, could you please review this PR if it's ready for merging?

**Comment by dotlambda** [comment]
Please squash and update the PR title.

File: pkgs/applications/video/mpv/scripts/memo.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+  meta = with lib; {
+    description = "A recent files menu for mpv";
+    homepage = "https://github.com/po5/memo";
+    license = licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ purrpurrn ];
+  };
+}
+
+
```
Comment (dotlambda) [inline pkgs/applications/video/mpv/scripts/memo.nix:38]: Remove unnecessary lines.

File: pkgs/applications/video/mpv/scripts/memo.nix [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  pname = "memo";
+  version = "0-unstable-2023-10-31";
+
+  src = fetchFromGitHub {
+    owner = "po5";
+    repo = "memo";
+    rev = "6f2b036ef860e130ea584657389e0a7ceb4d954f";
+    hash = "sha256-m8ikXuw7PM4Btg8w7ufLneKA4fnYjMyfVJYueZILMw8=";
+  };
+  passthru.updateScript = unstableGitUpdater {};
```
Comment (dotlambda) [inline pkgs/applications/video/mpv/scripts/memo.nix:17]:
```suggestion

  passthru.updateScript = unstableGitUpdater {};
```

File: pkgs/applications/video/mpv/scripts/memo.nix [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+  passthru.scriptName = "memo.lua";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/mpv/scripts
+    cp -r memo.lua $out/share/mpv/scripts
```
Comment (dotlambda) [inline pkgs/applications/video/mpv/scripts/memo.nix:25]: Please use `install -Dt`.

Comment (purrpurrn) [reply pkgs/applications/video/mpv/scripts/memo.nix:25]: it seems like `installPhase` and `passthru.scriptName` is completely redundant.
```suggestion
  passthru.scriptName = "memo.lua";

  installPhase = ''
    runHook preInstall

      install -Dt memo.lua $out/share/mpv/scripts

    runHook postInstall
  '';
```

Comment (dotlambda) [reply pkgs/applications/video/mpv/scripts/memo.nix:26]: Even `passthru.scriptName` is not needed because if will be computed from `pname`.

Comment (purrpurrn) [reply pkgs/applications/video/mpv/scripts/memo.nix]: removed both :+1:

**Comment by purrpurrn** [comment]
> Please squash and update the PR title.

wdym about the PR title? Should I append a 0- to the start?

**Comment by dotlambda** [comment]
It would be nicer if your commit message didn't contain that gibberish.
I'm going to wait a bit to see if someone familiar with mpc scripts in nixpkgs reacts. If not I'll merge.

**Comment by purrpurrn** [comment]
> It would be nicer if your commit message didn't contain that gibberish. I'm going to wait a bit to see if someone familiar with mpc scripts in nixpkgs reacts. If not I'll merge.

Doesn't seem like

> It would be nicer if your commit message didn't contain that gibberish. I'm going to wait a bit to see if someone familiar with mpc scripts in nixpkgs reacts. If not I'll merge.

Nobody reviewed after 3 weeks, can this PR get merged? (Sorry, I will do better in the future when PRing)

**Comment by purrpurrn** [comment]
@dotlambda Can you review this again?

**Review by nicoonoclaste — APPROVED** [review]
LGTM, though I will try and test this later today.
@purrpurrn, if that's fine with you I'll rebase this and handle the merge conflict before merging; otherwise, you can do it yourself and tag me once it's done.

> Nobody reviewed after 3 weeks, can this PR get merged?

Sorry, I was tagged for a review but wasn't really active the last months due to health complications.  :(

**Comment by purrpurrn** [comment]
I hope you are recovering or recovered from your health complications!

@nbraud.

**Pull request merged**